src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationOptionsProviderExtension.cs

```diff
@@ -38,9 +42,14 @@ public static class CompilationOptionsProviderExtension
                 throw new ArgumentNullException(nameof(applicationEnvironment));
             }
 
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
             return compilerOptionsProvider.GetCompilerOptions(applicationEnvironment.ApplicationName,
```

Comment:
Completely unrelated - while you're modifying this type, could you reformat this bit of code so it uses the VS default indentation:
```C#
return compilerOptionsProvider.GetCompilerOptions(
        applicationEnvironment.ApplicationName,
        applicationEnvironment.RuntimeFramework,
        configuration)
    .ToCompilationSettings(applicationEnvironment.RuntimeFramework, applicationEnvironment.ApplicationBasePath);
```